chrome/content/zotero/xpcom/style.js

```diff
@@ -190,7 +190,26 @@ Zotero.Styles = new function() {
 		}
 		
 		return _styles[id] || false;
-	}
+	};
+	
+	this.getAsync = Zotero.Promise.coroutine(function* (id, skipMappings, download=true) {
+		yield this.init();
+		var style = this.get(id, skipMappings);
+		
+		if (!style && download) {
+			try {
+				// Don't wait too long
+				Zotero.debug(`Fetching required style ${id} asynchronously`);
+				let styleText = yield Zotero.File.getContentsFromURLAsync(id, {timeout: 5000});
```

Comment:
We <a href="http://docs.citationstyles.org/en/stable/specification.html#info">should be</a> checking for a `rel="self"` first, falling back to the `id`. (The whole id situation in CSL is non-ideal — the `id` would ideally be non-changing, with a (changeable) `rel="self"` for download (in which case renamed-styles.json might still be used for creating server-side `rel="link"` redirects but wouldn't be necessary for client-side resolution. But that's a discussion for elsewhere.)

/cc @rmzelle